aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationLinkedList.java

```diff
@@ -0,0 +1,112 @@
+package com.amplifyframework.datastore.syncengine;
+
+import com.amplifyframework.core.model.Model;
+import java.util.HashMap;
+
+/*
+ * The {@link MutationLinkedList} is a LinkedHashMap like container , the goal of using this container is to
+ * achieve O(1) time complexity for both getting a {@link PendingMutation} and update an existing mutation with
+ * valid id.
+ */
+public class MutationLinkedList {
```

Comment:
Should you implement any of the standard Java contracts? `extends LinkedHashMap`,  `implements List`, ... `implements Comparable<T>`, implements `Collection<T>`, etc.?

------------

Also for name, I'd use `MutationQueue`. The `PersistentMutationOutbox` documentation should be updated to desribe how it differs from a `MutationQueue` -- that is, the outbox applies (what?) rules about populating the queue, etc.

--------

This new class should have a corresponding `MutationQueueTest.java` which tests it as an isolated unit.

Reply:
@jamesonwilliams will add some tests for this new class. I didnt implement collections' interface methods as they are not needed at MutationOutBox's use case, but i can add them